sporadic macOS 11 CI failing to launch on GitHub Actions with kLSNoExecutableErr

Starting around 2021-10-14 when GitHub Actions migrated my org from macOS from 10.15.7 to 11.6, I am getting very frequent errors trying to install Sublime:

```
2021-10-15T06:31:12.1951910Z copy the package to sublime package directory
2021-10-15T06:31:12.8262420Z cloning https://github.com/SublimeText/UnitTesting tag: 1.5.5 into /Users/runner/Library/Application Support/Sublime Text/Packages/UnitTesting...
2021-10-15T06:31:13.3746180Z 61a20f197224fb2532e85bd5e6d2193a9ca5e39d
2021-10-15T06:31:13.7309720Z cloning https://github.com/codexns/sublime-coverage tag: 1.0.0 into /Users/runner/Library/Application Support/Sublime Text/Packages/coverage...
2021-10-15T06:31:14.3511920Z bdda8348acf163fbb2c7e276d7d5415c2d0fa4bb
2021-10-15T06:31:14.3671080Z installing sublime text 4
2021-10-15T06:31:14.8501400Z downloading https://download.sublimetext.com/sublime_text_build_4113_mac.zip
2021-10-15T06:31:14.8674640Z   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2021-10-15T06:31:14.8675780Z                                  Dload  Upload   Total   Spent    Left  Speed
2021-10-15T06:31:14.8676170Z 
2021-10-15T06:31:15.7237410Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2021-10-15T06:31:16.6763680Z   3 41.5M    3 1679k    0     0  1951k      0  0:00:21 --:--:--  0:00:21 1960k
2021-10-15T06:31:16.6966370Z  94 41.5M   94 39.4M    0     0  21.7M      0  0:00:01  0:00:01 --:--:-- 21.7M
2021-10-15T06:31:16.7067900Z 100 41.5M  100 41.5M    0     0  22.7M      0  0:00:01  0:00:01 --:--:-- 22.8M
2021-10-15T06:31:16.7471240Z Archive:  /Users/runner/Downloads/sublimetext.zip
.... a bunch of inflating
2021-10-15T06:31:18.2387740Z creating sublime package directory
2021-10-15T06:31:18.3718710Z The application cannot be opened for an unexpected reason, error=Error Domain=NSOSStatusErrorDomain Code=-10827 "kLSNoExecutableErr: The executable is missing" UserInfo={_LSLine=3845, _LSFunction=_LSOpenStuffCallLocal}
2021-10-15T06:31:18.3754810Z ##[error]Process completed with exit code 1.
2021-10-15T06:31:18.4211580Z Cleaning up orphan processes
```

I don't know exactly what this means. I know macOS 11 introduced a lot of security changes (as almost every macOS release does).  I have seen a few other people around the internet experiencing similar issues, and a workaround is noted here:

https://stackoverflow.com/questions/66495254/opening-app-on-macos-11-big-sur-from-javafx-application-randomly-fails-with-klsn

I'm wondering if something similar might help here?

--- a/sbin/install_sublime_text.sh
+++ b/sbin/install_sublime_text.sh
@@ -104,6 +104,12 @@ if [ $(uname) = 'Darwin'  ]; then
             echo '{"close_windows_when_empty": false }' > "$STP/User/Preferences.sublime-settings"
         fi
 
+        # Workaround for an issue on macOS 11 where opening the app would
+        # randomly fail with kLSNoExecutableErr.
+        # See https://github.com/SublimeText/UnitTesting/issues/200
+        /System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Support/lsregister \
+            -f "$HOME/Applications/$SUBLIME_TEXT.app"
+
         # make `subl` available
         open "$HOME/Applications/$SUBLIME_TEXT.app"
         sleep 2

--- a/scripts/install_sublime_text.sh
+++ b/scripts/install_sublime_text.sh
@@ -105,6 +105,12 @@ if [ $(uname) = 'Darwin'  ]; then
             echo '{"close_windows_when_empty": false }' > "$STP/User/Preferences.sublime-settings"
         fi
 
+        # Workaround for an issue on macOS 11 where opening the app would
+        # randomly fail with kLSNoExecutableErr.
+        # See https://github.com/SublimeText/UnitTesting/issues/200
+        /System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Support/lsregister \
+            -f "$HOME/Applications/$SUBLIME_TEXT.app"
+
         # make `subl` available
         open "$HOME/Applications/$SUBLIME_TEXT.app"
         sleep 2

